contour-lint/src/test/java/com/squareup/contour/lint/NestedContourLayoutsDetectorTest.kt

```diff
@@ -0,0 +1,447 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.squareup.contour.lint
+
+import com.android.tools.lint.checks.infrastructure.LintDetectorTest.java
+import com.android.tools.lint.checks.infrastructure.LintDetectorTest.kotlin
+import com.android.tools.lint.checks.infrastructure.TestLintTask.lint
+import org.junit.FixMethodOrder
+import org.junit.Test
+import org.junit.runners.MethodSorters.JVM
+
+@Suppress("UnstableApiUsage")
+@FixMethodOrder(JVM)
+class NestedContourLayoutsDetectorTest {
+
+  companion object {
+    private val AXIS_SOLVERS_STUB = kotlin(
+        """
+        |package com.squareup.contour.solvers
+        |
+        |interface AxisSolver
+        |interface XAxisSolver : AxisSolver
+        |interface YAxisSolver : AxisSolver
+        """.trimMargin()
+    )
+    private val CONTOUR_LAYOUT_STUB = kotlin(
+        """
+        |package com.squareup.contour
+        |
+        |import android.content.Context
+        |import android.view.ViewGroup
+        |import com.squareup.contour.solvers.XAxisSolver
+        |import com.squareup.contour.solvers.YAxisSolver
+        |
+        |open class ContourLayout(context: Context) : ViewGroup(context) {
+        |  fun <T> T.layout(x: XAxisSolver, y: YAxisSolver): T = TODO()
+        |  fun <T> T.layout(addToViewGroup: Boolean, config: T.() -> LayoutSpec): T = TODO()
+        |  fun <T> T.applyLayout(x: XAxisSolver, y: YAxisSolver): T = TODO()
+        |}
+        """.trimMargin()
+    )
+    private val VIEW_STUB = java(
+        """
+        |package android.view;
+        |
+        |import android.content.Context;
+        |
+        |public class View {
+        |  public View(Context context) {}
+        |  public void layout(int l, int t, int r, int b) {}
+        |}
+        """.trimMargin()
+    )
+  }
```

Comment:
nit: move companion object to bottom of source file

Reply:
done